src/thing.h

```diff
@@ -4,14 +4,37 @@
 #ifndef FS_THING_H
 #define FS_THING_H
 
+#include "enums.h"
+
 class Container;
 class Creature;
-class Cylinder;
 class Item;
 class Tile;
 struct Position;
 
-class Thing
+inline constexpr int32_t INDEX_WHEREEVER = -1;
+
+enum cylinderflags_t
+{
+	FLAG_NOLIMIT = 1 << 0,             // Bypass limits like capacity/container limits, blocking items/creatures etc.
+	FLAG_IGNOREBLOCKITEM = 1 << 1,     // Bypass movable blocking item checks
+	FLAG_IGNOREBLOCKCREATURE = 1 << 2, // Bypass creature checks
+	FLAG_CHILDISOWNER = 1 << 3,        // Used by containers to query capacity of the carrier (player)
+	FLAG_PATHFINDING = 1 << 4,         // An additional check is done for floor changing/teleport items
+	FLAG_IGNOREFIELDDAMAGE = 1 << 5,   // Bypass field damage checks
+	FLAG_IGNORENOTMOVEABLE = 1 << 6,   // Bypass check for mobility
+	FLAG_IGNOREAUTOSTACK = 1 << 7,     // queryDestination will not try to stack items together
+};
+
+enum cylinderlink_t
+{
+	LINK_OWNER,
+	LINK_PARENT,
+	LINK_TOPPARENT,
+	LINK_NEAR,
+};
```

Comment:
Renaming enums to `ReceiverFlags` and `ReceiverLink`, would make their purpose clearer and consistent with new naming direction